opennms-doc/guide-admin/src/asciidoc/text/performance-data-collection/property-extenders.adoc

```diff
@@ -0,0 +1,487 @@
+
+[source, text]
+----
+bsnAPIfLoadParametersEntry OBJECT-TYPE
+    -- ...
+    DESCRIPTION
+        "An entry (conceptual row) in the Table.
+        Entries in this MIB are indexed by
+        bsnAPDot3MacAddress and bsnAPIfSlotId"
+    INDEX           {
+                        bsnAPDot3MacAddress,
+                        bsnAPIfSlotId
+                    } <1>
+
+    -- ...
+
+bsnAPEntry OBJECT-TYPE
+    -- ...
+    DESCRIPTION
+        "An entry in the bsnAPTable."
+    INDEX           { bsnAPDot3MacAddress } <2>
+    -- ...
+----
+<1> `bsnAPDot3MacAddress` is the first component of the compound index for the entry type for `bsnAPIfLoadParametersTable`
+<2> `bsnAPDot3MacAddress` is the sole index for the entry type for `bsnAPTable`
+
+By extracting just the first index component and using the result as an index into the source MIB table, it's possible to import the human-sensible `bsnAPName` string property from the source MIB table.
+
+===== Configuring the Regex Property Extender
+The Regex property extender expects three parameters, all of which are required:
+
+.Regex Property Extender Parameters
+[options="header, autowidth"]
+|===
+| Name | Description
+| `source-type` | The name of the `resourceType` associated with the source MIB table
+| `source-alias` | The alias name of the string property to be imported from the source MIB table
+| `index-pattern` | A regular expression containing one matching group
+|===
+
+The `index-pattern` expression must meet the same criteria as for the Index Split property extender.
+The subpattern matched by its first capturing group will be used as an index into the source MIB table; any further groups are ignored.
+
+This example shows how to use the value of `bsnAPDot3MacAddress` as an index into the `bsnAPTable`.
+
+[source, xml]
+----
+<resourceType name="bsnAPEntry" label="Cisco Wireless AP" resourceLabel="${bsnAPName} (index ${index})">
+  <persistenceSelectorStrategy class="org.opennms.netmgt.collection.support.PersistAllSelectorStrategy" />
+  <storageStrategy class="org.opennms.netmgt.collection.support.IndexStorageStrategy" />
+</resourceType>
+
+<resourceType name="bsnAPIfLoadParametersEntry" label="Cisco Wireless AP Resources" resourceLabel="${bsnAPName} (index ${index})">
+  <persistenceSelectorStrategy class="org.opennms.netmgt.collection.support.PersistAllSelectorStrategy" />
+  <storageStrategy class="org.opennms.netmgt.collection.support.IndexStorageStrategy" />
+</resourceType>
+
+<groups>
+  <group name="bsnAPTable" ifType="all">
+    <mibObj oid=".1.3.6.1.4.1.14179.2.2.1.1.3" instance="bsnAPEntry" alias="bsnAPName" type="string" /> <1>
+  </group>
+
+  <group name="bsnAPIfLoadParametersTable" ifType="all">
+    <mibObj oid=".1.3.6.1.4.1.14179.2.2.13.1.4" instance="bsnAPIfLoadParametersEntry" alias="bsnAPIfLoadNumOfCli" type="integer" />
+    <property instance="bsnAPIfLoadParametersEntry" alias="bsnAPName"> <2>
```

Comment:
I don't recall if the Regex Extender is the default one, but I think the docs should explicitly include the `class-name` for each `property` to avoid confusion.

Reply:
It is the default, but I agree all the doc examples need to be explicit. Thanks for catching that.